### PR TITLE
fix recorded used semantics in decoding

### PR DESCRIPTION
**Pull request by piotr-rudnik**

### CODE REVIEW

File: ethtx/providers/semantic_providers/semantics_repository.py [hunk 1]
```diff
@@ -254,6 +254,9 @@ def get_semantics(self, chain_id: str, address: str) -> Optional[AddressSemantic
         # amend semantics with locally stored updates
         amend_contract_semantics(address_semantics.contract)
 
+        if self._records is not None:
+            self._records.append(address)
```
Comment (kchojn) [inline ethtx/providers/semantic_providers/semantics_repository.py:258]: If an error occurs before the `end_record()` method, how `self._records` will be cleared?

File: ethtx/decoders/decoder_service.py [hunk 2]
```diff
@@ -9,19 +9,23 @@
...
+log = logging.getLogger(__name__)
 
 class DecoderService:
-    def __init__(self, abi_decoder, semantic_decoder, web3provider, default_chain):
-        self.abi_decoder = abi_decoder
-        self.semantic_decoder = semantic_decoder
+    def __init__(self, abi_decoder: ABIDecoder, semantic_decoder: SemanticDecoder, web3provider, default_chain: str):
+        self.abi_decoder: ABIDecoder = abi_decoder
+        self.semantic_decoder: SemanticDecoder = semantic_decoder
         self.web3provider = web3provider
```
Comment (kchojn) [inline ethtx/decoders/decoder_service.py:26]: is it possible to get `web3provider` type?

File: ethtx/providers/semantic_providers/semantics_repository.py (hunk 1 above)
Comment (piotr-rudnik) [reply ethtx/providers/semantic_providers/semantics_repository.py:258]: self._record is replaced with every decode so it shouldn't be a problem

File: ethtx/decoders/decoder_service.py (hunk 2 above)
Comment (piotr-rudnik) [reply ethtx/decoders/decoder_service.py]: added

File: ethtx/decoders/decoder_service.py [hunk 3]
```diff
@@ -16,14 +16,15 @@
 from .semantic.decoder import SemanticDecoder
 from ..models.decoded_model import DecodedTransaction
 from ..models.objects_model import Block, Call
+from ..providers.web3_provider import NodeDataProvider
 
 log = logging.getLogger(__name__)
 
 class DecoderService:
     def __init__(self, abi_decoder: ABIDecoder, semantic_decoder: SemanticDecoder, web3provider, default_chain: str):
```
Comment (kchojn) [inline ethtx/decoders/decoder_service.py:24]: and here pls

**Review by kchojn — APPROVED** [review]

**Pull request merged**